src/FakeIO.java

```diff
@@ -0,0 +1,19 @@
+import java.util.LinkedList;
+import java.util.List;
+
+public class FakeIO implements IOInterface {
+   private LinkedList<Integer> numbers;
+
+    public FakeIO(List<Integer> guessedNumbers) {
+        this.numbers = new LinkedList<>(guessedNumbers);
+    }
+
+    @Override
+    public int getInput() {
+        return numbers.pop();
+    }
+
```

Comment:
Quite a few extra lines between the last method and the end of the class :joy_cat: